lib/cw-core/src/util.rs

```diff
@@ -57,3 +57,323 @@ pub fn split_framebuffer(framebuffer: &[u8]) -> Vec<Vec<u8>> {
 
     framebuffers
 }
+
+pub fn get_different_framebuffer(previous_framebuffer: &mut [u8], framebuffer: &[u8]) -> Vec<u8> {
+    //handle incomparable frambuffer
+    if previous_framebuffer.len() != framebuffer.len() {
+        return vec![0]; //error
+    }
+
+    //handle invalid buffer
+    if previous_framebuffer.len() % 3 != 0 {
+        return vec![0];
+    }
+
+    let mut different_framebuffer_vector: Vec<u8> = vec![];
+
+    let frame_length = previous_framebuffer.len();
+    for i in (0..frame_length - 2).step_by(3) {
+        let start_i: usize = i;
+        let end_i: usize = i + 3;
+
+        //get the previous and the current rgb code
+        let prev_rgb: &[u8] = &previous_framebuffer[start_i..end_i];
+        let cur_rgb: &[u8] = &framebuffer[start_i..end_i];
+
+        if prev_rgb != cur_rgb {
+            //add index and updated rgb code to the different buffer
+            different_framebuffer_vector.extend(get_pixel_index(i));
+            different_framebuffer_vector.extend(cur_rgb);
+
+            //update the previous buffer to the latest rgb code
+            previous_framebuffer[start_i..end_i].copy_from_slice(cur_rgb);
+        }
+    }
+
+    return different_framebuffer_vector;
+}
+
+pub fn get_pixel_index(index: usize) -> [u8; 2] {
+    if index % 3 != 0 {
```

Comment:
Something has gone horribly wrong if this fails, to please just let it crash at that point. Easier to debug

`assert_eq!(index % 3, 0, "Your error message");`